packages/shared/src/hooks/squads/usePublicSquadRequests.ts

```diff
@@ -1,22 +1,61 @@
-import { useMutation } from '@tanstack/react-query';
+import {
+  InfiniteData,
+  useInfiniteQuery,
+  useMutation,
+} from '@tanstack/react-query';
+import { subDays } from 'date-fns';
 import { useToastNotification } from '../useToastNotification';
-import { submitSquadForReview } from '../../graphql/squads';
-import { ApiErrorResult } from '../../graphql/common';
+import {
+  getPublicSquadRequests,
+  submitSquadForReview,
+} from '../../graphql/squads';
+import { ApiErrorResult, Connection } from '../../graphql/common';
 import { parseOrDefault } from '../../lib/func';
-import { PublicSquadRequest } from '../../graphql/sources';
+import {
+  PublicSquadRequest,
+  PublicSquadRequestStatus,
+} from '../../graphql/sources';
+import { generateQueryKey, RequestKey, StaleTime } from '../../lib/query';
+import { useAuthContext } from '../../contexts/AuthContext';
+import { SquadStatus } from '../../components/squads/settings';
 
 const DEFAULT_ERROR = "Oops! That didn't seem to work. Let's try again!";
 
 interface UsePublicSquadRequestsResult {
-  submitForReview: (squadId: string) => Promise<PublicSquadRequest>;
+  submitForReview: () => Promise<PublicSquadRequest>;
   isSubmitLoading: boolean;
+  requests: InfiniteData<Connection<PublicSquadRequest>>;
+  latestRequest: PublicSquadRequest;
+  isFetched: boolean;
+  status: SquadStatus;
 }
 
-export const usePublicSquadRequests = (): UsePublicSquadRequestsResult => {
+interface UsePublicSquadRequestsProps {
+  isQueryEnabled?: boolean;
+  sourceId: string;
+  status?: string;
+}
+
+const remoteStatusMap: Record<PublicSquadRequestStatus, SquadStatus> = {
+  [PublicSquadRequestStatus.Approved]: SquadStatus.Approved,
```

Comment:
why do we have 2 of these?

Reply:
One is remote values (API), one is from the frontend. The API doesn't have `InProgress`.